bin/yarn2nix.js

```diff
@@ -23,46 +24,70 @@ Options:
 `
 
 const HEAD = `
-{fetchurl, linkFarm}: rec {
+{fetchurl, linkFarm, fetchgit, runCommandNoCC, gnutar}: rec {
   offline_cache = linkFarm "offline" packages;
   packages = [
 `.trim();
 
 ////////////////////////////////////////////////////////////////////////////////
 
-function generateNix(lockedDependencies) {
-  let found = {};
-
-  console.log(HEAD)
-
-  for (var depRange in lockedDependencies) {
-    let dep = lockedDependencies[depRange];
+async function fetchgit(url, sha1) {
+  let file_name = path.basename(url);
+  let url_for_git = url.replace(/^git\+/, "")
+  const {error, stdout, stderr} = await execFile("nix-prefetch-git", [url_for_git, sha1], {});
```

Comment:
you can't call `nix-prefetch-git` during build phase https://github.com/NixOS/nixpkgs/issues/43093

I've tried and got an error 

```
(node:27308) UnhandledPromiseRejectionWarning: Error: Command failed: nix-prefetch-git https://github.com/srghma/node-shell-quote.git 0aa381896e0cd7409ead15fd444f225807a61e0a
Initialized empty Git repository in /tmp/nix-build-yarn.nix.drv-0/git-checkout-tmp-22QqXk01/node-shell-quote-0aa3818/.git/
From https://github.com/srghma/node-shell-quote
 * branch            newbranch  -> FETCH_HEAD
 * [new branch]      newbranch  -> origin/newbranch
Switched to a new branch 'fetchgit'
removing `.git'...
error: cannot open connection to remote store 'daemon': reading from file: Connection reset by peer
````